src/autotype/AutoTypeAction.cpp

```diff
@@ -91,3 +113,52 @@ void AutoTypeExecutor::execClearField(AutoTypeClearField* action)
 {
     Q_UNUSED(action);
 }
+
+void AutoTypeExecutor::execPickChars(AutoTypePickChars* action)
+{
+    // save active window
+    WId previousWindow = action->plugin->activeWindow();
+
+    // create and show char picker dialog
+    AutoTypePickCharsDialog* pickCharsDialog = new AutoTypePickCharsDialog(action->string);
+#if defined(Q_OS_MAC)
+    action->plugin->raiseOwnWindow();
+    Tools::wait(500);
+#endif
+    pickCharsDialog->show();
+    pickCharsDialog->activateWindow();
+
+    // wait for dialog to finish
+    QEventLoop loop;
+    QObject::connect(pickCharsDialog, SIGNAL(accepted()), &loop, SLOT(quit()));
+    QObject::connect(pickCharsDialog, SIGNAL(rejected()), &loop, SLOT(quit()));
```

Comment:
You do not check whether dialog was accepted or rejected. A rejected dialog should cause the auto-type sequence to be canceled right here.